### ISSUE TITLE
Cannot update ImageField to a new image file (Wagtail 1.3.1, Python 3.4.4)

### ISSUE THREAD

**Issue by Proper-Job** [torchbox/wagtail#2181]
When I initially save an image to an ImageField that is a subclass of Page the image is properly saved.
An update to that image sets the ImageField's path to an invalid path and doesn't upload the file.

Steps to reproduce:
```shell
wagtail start image_field_test
cd image_field_test/
```

Change home/model.py definition to :
```python
from __future__ import unicode_literals
from django.db import models
from wagtail.wagtailcore.models import Page
from wagtail.wagtailadmin.edit_handlers import FieldPanel

class HomePage(Page):
    image = models.ImageField(upload_to='uploaded_images', default='')

    content_panels = Page.content_panels + [
        FieldPanel('image'),
    ]
```

```shell
mkdir -p media/uploaded_images
chmod -R 777 media
python manage.py makemigrations
python manage.py migrate
python manage.py createsuperuser
python manage.py runserver
```

-  Log into the admin interface and navigate to the home page (currently http://127.0.0.1:8000/admin/pages/3/edit/).
-  Add an image and publish
-  Go back to the page and notice how the image path is missing the uploaded_images prefix and clicking on the link goes nowhere.  The file is also not present in the file system.
- When creating a new home page I can add an image once.  On update I get the problem detailed above.

**Comment by bhunter** [comment]
I'm having a similar issue with models.FileField. Despite setting upload_to='images/' there is no uploaded file and when the model's FileField is checked the path does not reflect the MEDIA_ROOT/images path.
Migrations show that the field has it's upload_to value set.

**Comment by rafadev** [comment]
I'm detecting the same problem, its pretty serious as it renders the ImageField unusable.

When I create a page containing an ImageField, the first time its saved it works fine, however, if I change the image and save it again, the file isn't uploaded at all. The same thing happens regardless if I specify upload_to on the field.

**Comment by thecybil** [comment]
This issue also occurs with snippets (wagtail 1.4.1, Python 2.7.10).

It happens because the page edit template isn't properly setting the form enctype to "multipart/form-data".

A workaround which worked for me was overriding the wagtailadmin/pages/edit.html and editing the line that looks like: 

```
    <form id="page-edit-form" action="{% url 'wagtailadmin_pages:edit' page.id %}" method="POST"{% if form.is_multipart %} enctype="multipart/form-data"{% endif %}>`
```

with

```
    <form id="page-edit-form" action="{% url 'wagtailadmin_pages:edit' page.id %}" method="POST" enctype="multipart/form-data">`
```

Similarly, you can override the following templates for wagtailsnippets:

* /wagtailsnippets/snippets/edit.html
* /wagtailsnippets/snippets/create.html

Replace:
```
 <form action="{% url 'wagtailsnippets:edit' model_opts.app_label model_opts.model_name instance.id %}" method="POST">
```

with 

```
 <form action="{% url 'wagtailsnippets:edit' model_opts.app_label model_opts.model_name instance.id %}" method="POST" enctype="multipart/form-data">
```

**Comment by timheap** [comment]
I can confirm the issue with uploading files / images to the Page edit screen. The files successfully upload when creating the page, but not when editing the page. I will look in to this.

@thecybil: The issue with snippets is a different issue - the `enctype` is missing in those forms as you have found. I've got a separate PR for that issue at #2420.

**Comment by timheap** [comment]
I've tracked down the issue to some code in django-modelcluster. I've fixed it there, created a PR (torchbox/django-modelcluster#53), and started a PR (#2422) to fix it in Wagtail once the django-modelcluster issue is fixed. Thanks for your bug reports!

**Comment by gasman** [comment]
Fixed in #2422 (and more specifically https://github.com/torchbox/django-modelcluster/pull/53)
